app/controllers/candidate_interface/apply_from_find_controller.rb

```diff
@@ -6,18 +6,12 @@ class ApplyFromFindController < CandidateInterfaceController
     skip_before_action :require_basic_auth_for_ui, if: -> { ENV['DISABLE_BASIC_AUTH_FOR_LANDING_PAGE'] }
 
     rescue_from ActionController::ParameterMissing, with: :render_not_found
+    rescue_from ActiveRecord::RecordNotFound, with: :render_not_found
 
     def show
-      provider_code = params.fetch(:providerCode)
-      course_code = params.fetch(:courseCode)
-
-      course = FindAPI::Course.fetch(provider_code, course_code)
-
-      if course.nil?
-        render_not_found
-      else
-        @course = CoursePresenter.new course
-      end
+      provider = Provider.find_by!(code: params.fetch(:providerCode))
+      course = provider.courses.where(exposed_in_find: true).find_by!(code: params.fetch(:courseCode))
```

Comment:
`.where(exposed_in_find: true)` feels like it could be a `.findable` scope, feels like we may reuse it. Minor